src/lightning/app/cli/connect/node.py

```diff
@@ -0,0 +1,177 @@
+        out = subprocess.run(
+            f"docker ps -q -f name={CODE_SERVER_CONTAINER}", shell=True, check=True, capture_output=True
+        )
+        if out.stdout:
+            pass
+        else:
+            out = subprocess.run(
+                f"docker container ls -aq -f name={CODE_SERVER_CONTAINER}", shell=True, check=True, capture_output=True
+            )
+            if out.stdout:
+                subprocess.run(f"docker rm -f {CODE_SERVER_CONTAINER}", shell=True, check=True)
+            else:
+                live.update(Spinner("point", text=Text("pulling code server image", style="white")))
+                out = subprocess.run(f"docker pull {CODE_SERVER_IMAGE}", shell=True, check=True, capture_output=True)
+                error = out.stderr
+                if error:
+                    live.stop()
+                    rich.print(f"[red]Failed[/red]: code server image pull failed with error: {str(error)}")
+                    return
+            cmd = get_code_server_docker_command()
+            live.update(Spinner("point", text=Text("running code server", style="white")))
+            _ = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        # if lightning daemon is already running, ignore.
+        # If not, but container exists, remove it and run. Otherwise, run.
+        out = subprocess.run(
+            f"docker ps -q -f name={LIGHTNING_DAEMON_CONTAINER}", shell=True, check=True, capture_output=True
+        )
+        if out.stdout:
+            pass
+        else:
+            out = subprocess.run(
+                f"docker container ls -aq -f name={LIGHTNING_DAEMON_CONTAINER}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            if out.stdout:
+                subprocess.run(f"docker rm -f {LIGHTNING_DAEMON_CONTAINER}", shell=True, check=True)
+            else:
+                live.update(Spinner("point", text=Text("pulling lightning daemon image", style="white")))
+                out = subprocess.run(
+                    f"docker pull {LIGHTNING_DAEMON_IMAGE}", shell=True, check=True, capture_output=True
+                )
+                error = out.stderr
+                if error:
+                    live.stop()
+                    rich.print(f"[red]Failed[/red]: lightnign daemon image pull failed with error: {str(error)}")
+                    return
+            cmd = get_lightning_daemon_command(name)
+            live.update(Spinner("point", text=Text("running lightning daemon", style="white")))
+            _ = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        # wait until if both docker containers are running
+        code_server_running = False
+        lightning_daemon_running = False
+        live.update(Spinner("point", text=Text("establishing connection ...", style="white")))
+        while True:
+            out = subprocess.run(
```

Comment:
You don't need this to register the node right ?

Reply:
If it is there for the demo, it is fine.

Reply:
So right now, "registration" is just in the asthetics. It immedietly starts the vscode server in local node. Without the vscode server, the proxy doesn't really have anything to do.

Reply:
Once we have cloudlet ready, vscode startup will be the responsibility of cloudlet, when the user triggers that in the UI or from the CLI